### ISSUE TITLE
Boot is delayed when drive is disconnected

### ISSUE THREAD

**Issue by aardbol** [systemd/systemd#11653]
240
> ...

**Used distribution**
 > Antergos (arch)

**Expected behaviour you didn't see**
 > PC boots without delays

**Unexpected behaviour you saw**
 > PC boot is delayed (1m30s) because systemd is waiting for a drive that is not connected to the PC

**Steps to reproduce the problem**
Shutdown the PC, disconnect the mofo hard drives that are too noisy to keep online all the time. Start up PC, experience the delay.

Before the last two updates of systemd, this issue wasn't the case

**Comment by kirbyfan64** [comment]
Does the fstab already have nofail? If not, you need to add it; I'm not sure why this wouldn't have come up before in that case, but it's been the behavior for a long time.

**Comment by yuwata** [comment]
Yeah, please specify nofail mount option.

Duplicate of #11614.

**Comment by IsaakGroepT** [comment]
nofail has always been part of the mount options in fstab for my secondary. As I said, before the last two version updates this wasn't an issue. And my fstab records for these drives is still the same as it was back then

**Comment by yuwata** [comment]
So, you mean nofail does not work?

**Comment by yuwata** [comment]
Then, please provide relevant fstab entry and generated .mount unit (maybe in /run/systemd/generator).

**Comment by yuwata** [comment]
Maybe, a duplicate of #11558. Please test #11562 or current git master.

**Comment by IsaakGroepT** [comment]
These three records are causing the delay to happen when I comment them out and they're not connected:

```
#UUID=1CFE4D2FFE4D030E /mnt/2TBNTFS ntfs-3g uid=1000,gid=users,dmask=027,fmask=137,locale=nl_BE.utf8,nofail 0 2
#UUID=e803918c-578c-43aa-b23b-52d7619b6247 /mnt/512GBPRO850 btrfs defaults,compress-force=zstd,space_cache,ssd,noatime,nofail 0 2
#UUID=c1657bcc-19c4-4d33-9e09-e31601e2c8d9 /mnt/2TBBTRFS btrfs defaults,compress-force=zstd,space_cache,noatime,autodefrag,nofail 0 2
```

I can't provide you with the .mount since they're not active records

**Comment by yuwata** [comment]
Do you have any relevant logs when these entries are enabled?

**Comment by poettering** [comment]
Please provide debug logs when this happens (i.e. boot with systemd.log_level=debug on the kernel cmdline)

**Comment by IsaakGroepT** [comment]
Boot log as requested.

[bootlog.txt](https://github.com/systemd/systemd/files/2847659/bootlog.txt)

**Comment by yuwata** [comment]
```
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=/dev/disk/by-uuid/F1C5-179B where=/boot/efi type=vfat makefs=no nofail=no noauto=no
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=/dev/disk/by-uuid/616a00c0-6b76-44ee-8397-8a2a03a6c437 where=/ type=btrfs makefs=no nofail=no noauto=no
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=/dev/disk/by-uuid/616a00c0-6b76-44ee-8397-8a2a03a6c437 where=/home type=btrfs makefs=no nofail=no noauto=no
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=/dev/disk/by-uuid/616a00c0-6b76-44ee-8397-8a2a03a6c437 where=/var type=btrfs makefs=no nofail=no noauto=no
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=tmpfs where=/ramdisk type=tmpfs makefs=no nofail=no noauto=no
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=/dev/disk/by-uuid/1CFE4D2FFE4D030E where=/mnt/2TBNTFS type=ntfs-3g makefs=no nofail=no noauto=yes
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Checking was requested for /dev/disk/by-uuid/1CFE4D2FFE4D030E, but fsck.ntfs-3g does not exist.
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=/dev/disk/by-uuid/b1615d2c-b467-461c-87bf-b89f466ef64c where=/mnt/275GBXFS type=xfs makefs=no nofail=no noauto=yes
feb 09 09:56:40 aardbol systemd-fstab-generator[408]: Found entry what=/dev/disk/by-uuid/b9c11660-b6a7-4d42-b684-a20a3256ea6d where=/mnt/SandiskSSD type=btrfs makefs=no nofail=no noauto=yes
feb 09 09:56:40 aardbol kernel: printk: systemd-fstab-g: 5 output lines suppressed due to ratelimiting
feb 09 09:56:40 aardbol systemd-gpt-auto-generator[410]: Waiting for device (parent + 2 partitions) to appear...
```
Hmm? You disabled `nofail` option?

**Comment by IsaakGroepT** [comment]
That's strange.. I haven't disabled it at all!

**Comment by yuwata** [comment]
Please paste /etc/fstab used when the above log is obtained.

**Comment by IsaakGroepT** [comment]
It's the same as before. I only commented out two of these lines

**Comment by poettering** [comment]
could you please provide the full fstab when this happens?

**Comment by IsaakGroepT** [comment]
This is my full fstab

```
UUID=F1C5-179B /boot/efi vfat defaults,noatime 0 0
UUID=616a00c0-6b76-44ee-8397-8a2a03a6c437 / btrfs defaults,noatime,compress=zstd,ssd,space_cache,subvol=_current/@ 0 1
UUID=616a00c0-6b76-44ee-8397-8a2a03a6c437 /home btrfs defaults,subvol=_current/@home 0 2
UUID=616a00c0-6b76-44ee-8397-8a2a03a6c437 /var btrfs defaults,subvol=_current/@var 0 2
tmpfs /ramdisk tmpfs defaults,size=100M 0 0

UUID=1CFE4D2FFE4D030E /mnt/2TBNTFS ntfs-3g uid=1000,gid=users,dmask=027,fmask=137,locale=nl_BE.utf8,nofail 0 2
UUID=b1615d2c-b467-461c-87bf-b89f466ef64c /mnt/275GBXFS xfs defaults,nofail 0 2
UUID=b9c11660-b6a7-4d42-b684-a20a3256ea6d /mnt/SandiskSSD btrfs defaults,compress=zstd,space_cache,ssd,noatime,nofail 0 0
UUID=06f4f0f9-eb1a-41e2-9ec7-33dc637cf32b /mnt/256GBBTRFS btrfs defaults,compress=zstd,space_cache,ssd,noatime,nofail 0 2
UUID=c1657bcc-19c4-4d33-9e09-e31601e2c8d9 /mnt/2TBBTRFS btrfs defaults,compress-force=zstd,space_cache,noatime,autodefrag,nofail 0 2

UUID=46af6e6b-02cd-4225-b404-4fda4587cd8f /mnt/256GBEVO850 btrfs defaults,compress=zstd,ssd,space_cache,noatime 0 2
UUID=46af6e6b-02cd-4225-b404-4fda4587cd8f /home/ik/Apps btrfs defaults,subvolid=257 0 2

UUID=d3e90578-c98c-47e1-aab8-33baf8ca2361 /tmp xfs defaults 0 0
```

**Comment by yuwata** [comment]
`fstab-generator` output wrong debug log. See #11747.
So, it seems at least parsing fstab entry is fine.
Then, please show the generated mount unit files in /run/systemd/generator.

**Comment by poettering** [comment]
hmm, are there any other services that pull in the offeding mount units? either explicitly, or implicitly, for example through RequiresMountsFor= or WorkingDirectory= or something similar? Might be worth doing "systemctl show" on the offending mount unit to see what pulls it in?

Note that local-fs.target is only one way to pull in a mount unit. If a path or mount unit is referenced by another unit that is pulled in, then a mount unit might be pulled in too.

**Comment by IsaakGroepT** [comment]
This issue is not present anymore in version 241. Drives are not plugged in while the fstab records are commented out (and further left unedited)

**Comment by poettering** [comment]
OK, if the issue is gone already, let's close this. Thanks for reporting back!

**Comment by IsaakGroepT** [comment]
Correction, it was fixed in v240.93. Just for the record in case it pops up again later

**Comment by IsaakGroepT** [comment]
Actually, it's not fixed. It reappeared, but I can't say how and why
